zk.js/src/utxo/program-utxo.ts

```diff
@@ -464,23 +577,28 @@ export async function decryptProgramUtxo({
     lightWasm,
     compressed,
     assetLookupTable,
-    pspId,
-    pspIdl,
-    utxoName,
+    owner,
+    ownerIdl,
+    type,
   });
+
   if (!decryptedProgramOutUtxo.value) {
-    return decryptedProgramOutUtxo as Result<ProgramUtxo | null, UtxoError>;
-  }
-  if (
-    decryptedProgramOutUtxo.value.outUtxo.publicKey ===
-    STANDARD_COMPRESSION_PUBLIC_KEY.toString()
-  ) {
-    const bs58Standard = bs58.encode(
-      STANDARD_COMPRESSION_PRIVATE_KEY.toArray(),
-    );
-    const bs5832 = bs58.encode(new Uint8Array(32).fill(1));
-    account = Account.fromPrivkey(lightWasm, bs58Standard, bs5832, bs5832);
+    return decryptedProgramOutUtxo as Result<
+      ProgramUtxo<PlaceHolderTData> | null,
+      UtxoError
+    >;
   }
+  /// TODO: double check if this is needed
```

Comment:
do we have a test that if we encrypt a utxo which has the standard public key as owner we decrypt it with the standard public key?

Reply:
I believe we don't decrypt the public key but infer it assuming it is the public key of the account which decrypts the utxo

Reply:
adding todo